include/lbann/layers/regularizers/batch_normalization.hpp

```diff
@@ -305,7 +305,7 @@ class batch_normalization : public regularizer_layer {
                                   get_prev_activations().LDim(),
                                   m_mean->Buffer(),
                                   m_var->Buffer(),
-                                  this->m_cudnn->get_stream());
+                                  El::GPUManager::Stream());
```

Comment:
Isn't it a little abusing to use Hydrogen as a GPU manager?

Reply:
An alternative approach is to implement a GPU manager that wraps around Hydrogen's GPU manager, similar to the old cuDNN manager. It seems easier to me to use Hydrogen's GPU manager though.

Reply:
I understand it's easier with the current design and there's nothing wrong with it, for now. So, I'm not advocating changing the interface.

However, since Hydrogen is a library for matrix data and computations, it doesn't seem to be a good idea to use it as a GPU resource manager too. In general, libraries should not own resources such as GPU device contexts and (externally visible) streams, but the main program should do. For example, cuDNN and CUBLAS need stream to be passed from their user code.